controllers/experiment.coffee

```diff
@@ -7,14 +7,26 @@ config = require '../config'
 
 overrides = {}
 
+registerViewEvents = (isOrganic, isBot, params, userId, timestamp) ->
+  (if isOrganic and not isBot
+    log.info "ASSIGN: #{userId} #{JSON.stringify params}"
+    Experiments.registerAssignment params, userId, timestamp
+  else Promise.resolve(null))
+  .then ->
+    unless isBot
+      log.info "VIEW: #{userId} #{JSON.stringify params}"
+      return Experiments.registerView params, userId, timestamp
+
 class ExperimentCtrl
   assign: (req) ->
-    userId = req.body.userId
-    userId ?= req.body.id # LEGACY
-    fromUserId = req.body.fromUserId
+    userId = req.body.userId and String req.body.userId
```

Comment:
should probably use `String` in conversions.coffee too. And we should ignore bot conversions (eg a bot may stay on a page for a minute, counting as an engaged play)